common/firmware/adc/adc_comms.cpp

```diff
@@ -0,0 +1,18 @@
+#include "common/firmware/adc_comms.hpp"
+
+#include "platform_specific_hal_conf.h"
+
+using namespace adc;
+/*
+ * Public Functions
+ */
+ADC::ADC(ADC_interface ADC_intf_instance1, ADC_interface ADC_intf_instance2)
+    : ADC_intf1(ADC_intf_instance1), ADC_intf2(ADC_intf_instance2) {}
+
+auto ADC::get_readings() -> adc::VoltageRead {
+    auto voltage_read = adc::VoltageRead{
+        .z_motor = static_cast<uint16_t>(adc_read_voltage_z_motor()),
```

Comment:
this IMO is where we should convert the ADC readings into voltages - `adc_read_z_motor() * FULLSCALE_VOLTAGE / ADC_FULLSCALE_OUTPUT`. It could also happen in c I suppose